eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/CDDL.hs

```diff
@@ -164,30 +153,57 @@ move_instantaneous_rewards_cert =
 
 move_instantaneous_reward :: Rule
 move_instantaneous_reward =
-  "move_instantaneous_reward"
-    =:= arr [a (int 0 / int 1), a $ mp [0 <+ asKey stake_credential ==> coin]]
+  comment
+    [str|The first field determines where the funds are drawn from.
+        |  0 denotes the reserves,
+        |  1 denotes the treasury.
+        |If the second field is a map, funds are moved to stake credentials.
+        |Otherwise, the funds are given to the other accounting pot.
+        |NOTE:
+        |  This has been safely backported to Shelley from Alonzo.
+        |]
+    $ "move_instantaneous_reward"
+      =:= arr
+        [ a (int 0 / int 1)
+        , a (smp [0 <+ asKey stake_credential ==> delta_coin] / coin)
+        ]
+
+delta_coin :: Rule
+delta_coin =
+  comment
+    [str|This too has been introduced in Shelley as a backport from Alonzo.
+        |]
+    $ "delta_coin" =:= VInt
 
 stake_credential :: Rule
-stake_credential =
-  "stake_credential"
-    =:= arr
-      [0, a addr_keyhash]
-    / arr [1, a scripthash]
+stake_credential = "stake_credential" =:= credential
+
+credential :: Rule
+credential = "credential" =:= arr [0, a addr_keyhash] / arr [1, a script_hash]
 
 pool_params :: Named Group
 pool_params =
-  "pool_params"
-    =:~ grp
-      [ "operator" ==> pool_keyhash
-      , "vrf_keyhash" ==> vrf_keyhash
-      , "pledge" ==> coin
-      , "cost" ==> coin
-      , "margin" ==> unit_interval
-      , "reward_account" ==> reward_account
-      , "pool_owners" ==> set addr_keyhash
-      , "relays" ==> arr [0 <+ a relay]
-      , "pool_metadata" ==> (pool_metadata / VNil)
-      ]
+  comment
+    [str|        pool_keyhash: operator       
+        |                coin: pledge         
+        |                coin: cost           
+        |       unit_interval: margin         
+        |   set<addr_keyhash>: pool_owners    
+        |           [* relay]: relays         
+        |pool_metadata / null: pool_metadata  
```

Comment:
Thanks! I will remove the two comment lines. But I didn't understand what others we have skipped like `vrf_keyhash`. :slightly_smiling_face: Can you explain?

Reply:
Oh, I just meant that since we don't have a comment to fields like "vrf_keyhash" which seem self-explanatory, in the same way, we could skip the comments for relays and pool_metadata, since they would just be duplicating the name without adding anything.